src/dos/dos_files.cpp

```diff
@@ -394,8 +394,7 @@ bool DOS_FindFirst(const char* search, FatAttributeFlags attr, bool fcb_findfirs
 	char dir[DOS_PATHLENGTH];char pattern[DOS_PATHLENGTH];
 	size_t len = strlen(search);
 
-	const bool is_root = (len > 2) && (search[len - 2] == ':') &&
-	                     (attr == FatAttributeFlags::Volume);
+	const bool is_root = attr.volume && (len > 2) && (search[len - 2] == ':');
```

Comment:
Just a question:

`attr == FatAttributeFlags::Volume`, is true if the all the other bits are false.

Where as the new `attr.volume` doesn't care about the other (non-volume) bits. 

So for example, A "volume" device that's marked system and hidden would fail in the prior equality check, but would pass in the replacement.

I /think/ the second is more correct - but just mentioning it.

Reply:
@FeralChild64, there are more of these comparisons that could be fixed (by making the only check the desired bit): `git grep "== FatAttributeFlags"` will reveal them.

Reply:
The problem is, the remaining code looks like this:

```C++
	if (attr == FatAttributeFlags::Volume) {
		dta.SetResult(discLabel, 0, 0, 0, FatAttributeFlags::Volume);
		return true;
	} else if (attr.volume && isRoot && !fcb_findfirst) {
		// something else
```

so I suppose the `attr == FatAttributeFlags::Volume` check was done this way on purpose. I start having doubts about change in `dos_files.cpp`- but I can't find any way to make the code misbehave.

Reply:
> I start having doubts about change in dos_files.cpp- but I can't find any way to make the code misbehave.

Good call; I guess if we knew how the original code operated, we could adjust these bit checks a bit closer.. but then, things are working very solid in DOSBox, so likely the original authors nailed it.

Reply:
I have reverted the change to `is_root` calculation - `attr` is actually a parameter received from the guest application, and the whole thing is some kind of hack. This should be cleaned up some day, but for now I consider it too risky.